crates/runtime/src/embeddings/vector_search.rs

```diff
@@ -896,60 +896,75 @@ impl VectorSearch {
-                    return Err(Error::CannotVectorSearchDataset {
-                        data_source: tbl.clone()
-                    });
-                };
+                async move {
+                    tracing::debug!("Running vector search for table {:#?}", tbl);
+
+                    // Only support one embedding column per table.
+                    let table_provider = self
+                        .df
+                        .get_table(&tbl)
+                        .await
+                        .ok_or(Error::DataSourcesNotFound {
+                            data_source: vec![tbl.clone()],
+                        })?;
+
+                    let Some(embedding_table) = get_embedding_table(&table_provider).await else {
+                        return Err(Error::CannotVectorSearchDataset {
+                            data_source: tbl.clone()
+                        });
+                    };
+
+                    let embedding_columns = embedding_table.get_embedding_columns();
+                    let Some(embedding_column) = embedding_columns.first() else {
+                        return Err(Error::NoEmbeddingColumns {
+                            data_source: tbl.clone(),
+                        });
+                    };
+
+                    if search_vectors.len() != 1 {
+                        return Err(Error::IncorrectNumberOfEmbeddingColumns {
+                            data_source: tbl.clone(),
+                            num_embeddings: search_vectors.len(),
+                        });
+                    }
+                    match search_vectors.first() {
+                        None => unreachable!(),
+                        Some(embedding) => {
+                            let result = self
+                                .individual_search(
+                                    &tbl,
+                                    embedding.clone(),
+                                    primary_keys,
+                                    embedding_column,
+                                    embedding_table.is_chunked(embedding_column),
+                                    additional_columns,
+                                    where_cond.as_ref(),
+                                    &keywords,
+                                    *limit,
+                                )
+                                .await?;
+                            Ok::<_, Error>((tbl.clone(), result))
+                        }
+                    }
+                }
+            }).collect::<Vec<_>>();
 
-                let embedding_columns = embedding_table.get_embedding_columns();
-                let Some(embedding_column) = embedding_columns.first() else {
-                    return Err(Error::NoEmbeddingColumns {
-                        data_source: tbl.clone(),
-                    });
-                };
+            let results = futures::future::join_all(search_futures).await;
```

Comment:
```suggestion
            let results = futures::future::try_join_all(search_futures).await?;
```

Could we use `try_join_all` here? It avoids needing to make a manual loop to pull out the errors later.